appcrashtracker/src/main/java/com/org/appcrashtracker/ExceptionHandler.java

```diff
@@ -0,0 +1,277 @@
+package com.org.appcrashtracker;
+
+import com.org.appcrashtracker.utils.ApplicationInfoUtils;
+import com.org.appcrashtracker.utils.BatteryUtils;
+import com.org.appcrashtracker.utils.NetworkUtils;
+import ohos.aafwk.ability.Ability;
+import ohos.aafwk.content.Intent;
+import ohos.aafwk.content.Operation;
+import ohos.app.Context;
+import ohos.app.Environment;
+import ohos.bundle.BundleInfo;
+import ohos.global.resource.NotExistException;
+import ohos.global.resource.ResourceManager;
+import ohos.global.resource.WrongTypeException;
+import ohos.hiviewdfx.HiChecker;
+import ohos.hiviewdfx.HiLog;
+import ohos.hiviewdfx.HiLogLabel;
+import ohos.system.DeviceInfo;
+import ohos.utils.zson.ZSONException;
+import ohos.utils.zson.ZSONObject;
+
+import java.io.*;
+import java.util.Locale;
+
+import static com.example.appcrashtracker.ResourceTable.*;
+import static com.org.appcrashtracker.utils.MemoryUtils.*;
+
+public class ExceptionHandler implements java.lang.Thread.UncaughtExceptionHandler {
+    private final Ability ability;
+    private ZSONObject jObjectData;
+    private String abilityName;
+    private final HiLogLabel label = new HiLogLabel(HiLog.LOG_APP, 0x00201, getClass().getName());
+    private boolean className = false;
+    private boolean message = false;
+    private boolean localizedMessage = false;
+    private boolean causes = false;
+    private boolean stackTraceBoolean = false;
+    private boolean brandName = false;
+    private boolean deviceName = false;
+    private boolean sdkVersion = false;
+    private boolean release = false;
+    private boolean height = false;
+    private boolean width = false;
+    private boolean appVersion = false;
+    private boolean tablet = false;
+    private boolean deviceOrientation = false;
+    private boolean screenLayout = false;
+    private boolean vmHeapSize = false;
+    private boolean allocatedVmSize = false;
+    private boolean vmMaxHeapSize = false;
+    private boolean vmFreeHeapSize = false;
+    private boolean nativeAllocatedSize = false;
+    private boolean batteryPercentage = false;
+    private boolean batteryCharging = false;
+    private boolean batteryChargingVia = false;
+    private boolean sdCardStatus = false;
+    private boolean internalMemorySize = false;
+    private boolean externalMemorySize = false;
+    private boolean internalFreeSpace = false;
+    private boolean externalFreeSpace = false;
+    private boolean packageName = false;
+    private boolean networkMode = false;
+    private boolean country = false;
+
+    public ExceptionHandler(Ability ability) throws NotExistException, WrongTypeException, IOException {
+        this.ability = ability;
+        abilityName = ability.getClass().getSimpleName();
+        ResourceManager resources = ability.getResourceManager();
+        String postUrl = ability.getString(String_url);
```

Comment:
This declaration and assignment can be removed and directly used in next line.

Reply:
Resolved in commit 00bd46e77851473750360a5a91d148105bf9950c